mesh/mesh.cpp

```diff
@@ -710,10 +710,11 @@ void Mesh::GetLocalTriToTetTransformation(
    locpm.SetSize(3, 3);
    for (int j = 0; j < 3; j++)
    {
-      const IntegrationPoint &vert = TetVert->IntPoint(tv[to[j]]);
-      locpm(0, j) = vert.x;
-      locpm(1, j) = vert.y;
-      locpm(2, j) = vert.z;
+      int k = to[j];
+      const IntegrationPoint &vert = TetVert->IntPoint(tv[j]);
```

Comment:
TetVert->IntPoint(tv[j])， tv[j] or tv[k]?

Reply:
Um, ping @v-dobrev

Reply:
Would be good to add a unit test to distinguish these two cases…

Reply:
This modification basically replaces the use of the orientation stored in `i` with its inverse. This change affects only cases where the orientation is not inverse of itself which for the case of triangles are orientations 2 and 4 which are inverses of each other. These two orientations are never used in the `Mesh::faces_info` because the orientation for element 1 is always 0 and the orientation for element 2 is always odd (the normal is flipped).

The new use case (where the need to use the inverse orientation became essential) is in the call to `Mesh::GetLocalFaceTransformation` in `BuildVdofToVdofMap` in `submesh_utils.cpp`. In that case, the orientation can be 2 or 4 since it comes from the method `Mesh::GetBdrElementAdjacentElement`.

Actually, there is a similar use case (`Mesh::GetLocalFaceTransformation` called with the result of `Mesh::GetBdrElementAdjacentElement`) in the previously existing method `Mesh::GetBdrElementTransformation` in the case of meshes with L2 nodes (e.g. periodic meshes). That may have been a bug that has not been observed in practice. To me it seems like there are also other possible problems in that code -- I'll make a separate issue about that.

Reply:
We fixed the methods `GetLocalTriToTetTransformation` and `GetLocalQuadToHexTransformation` but we should also take a look at and fix, if needed, the methods `GetLocalTriToWdgTransformation`, `GetLocalTriToPyrTransformation`, `GetLocalQuadToWdgTransformation`, and `GetLocalQuadToPyrTransformation`.

Reply:
Looking more into this change, I think that `GetLocalTriToTetTransformation` and `GetLocalQuadToHexTransformation` do not need to be changed. The assumption is that `i%64` is the orientation if the face as generated by the element relative to the face element.

The problem is that when we call it with the result of `Mesh::GetBdrElementAdjacentElement`, `i%64` is the orientation of the boundary element w.r.t. the face element (which here is the same as the face as generated by the element) and we actually need to provide the inverse of that -- the orientation of the face as generated by the element w.r.t the boundary element. This is because we want the local map returned by `Mesh::GetLocalFaceTransformation` to map from the local coordinate system of the boundary element to the local coordinate system of the element.

I will add a new method similar to `Mesh::GetBdrElementAdjacentElement` that returns the inverse orientation and use that when we want to call `Mesh::GetLocalFaceTransformation` for a boundary element.

Reply:
Commit implementing the last comment: 5180b3f92.